pkg/skaffold/kubernetes/manifest/image_pull_policy.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manifest
+
+import apimachinery "k8s.io/apimachinery/pkg/runtime/schema"
+
+type ResourceSelectorImagePullPolicy struct{}
+
+func NewResourceSelectorImagePullPolicy() *ResourceSelectorImagePullPolicy {
+	return &ResourceSelectorImagePullPolicy{}
+}
+
+func (rs *ResourceSelectorImagePullPolicy) allowByGroupKind(gk apimachinery.GroupKind) bool {
+	if _, allowed := TransformAllowlist[gk]; allowed {
+		if rf, disallowed := TransformDenylist[gk]; disallowed {
+			for _, s := range rf.PodSpec {
+				if s == ".*" {
+					return false
+				}
+			}
+		}
+		return true
+	}
+	return false
+}
+
+func (rs *ResourceSelectorImagePullPolicy) allowByNavpath(gk apimachinery.GroupKind, navpath string, k string) (string, bool) {
+	if rf, ok := TransformDenylist[gk]; ok {
+		for _, denypath := range rf.PodSpec {
+			if denypath == ".*" || navpath == denypath {
+				return "", false
+			}
+		}
+	}
+	if rf, ok := TransformAllowlist[gk]; ok {
+		for _, allowpath := range rf.PodSpec {
+			if allowpath == ".*" || navpath == allowpath {
+				return "", true
+			}
+		}
+	}
+	return "", false
+}
+
+func (l *ManifestList) ReplaceImagePullPolicy(rs ResourceSelector) (ManifestList, error) {
+	r := &imagePullPolicyReplacer{}
+	return l.Visit(r, rs)
+}
+
+type imagePullPolicyReplacer struct{}
+
+func (i *imagePullPolicyReplacer) Visit(gk apimachinery.GroupKind, navpath string, o map[string]interface{}, k string, v interface{}, rs ResourceSelector) bool {
```

Comment:
Please add a comment explaining this method also. For example Visit sets the value of the "imagePullPolicy" field in a Kubernetes manifest to "Never".